Agents/PropertyValueEstimationAgent/agent/kg_operations/kgclient.py

```diff
@@ -0,0 +1,112 @@
+logger = agentlogging.get_logger('prod')
+
+
+class KGClient(PySparqlClient):
+    #
+    # SPARQL QUERIES
+    #
+    def get_transaction_details(self, tx_iri:str) -> dict:
+        # Retrieve transaction details for given transaction IRI
+        # Returns dictionary with keys: price, date, property_iri
+
+        query = f"""
+            SELECT ?price ?date ?property_iri
+            WHERE {{   
+            <{tx_iri}> <{LRPPI_DATE}> ?date ; 
+                       <{LRPPI_PRICE}> ?price ; 
+                       ^<{OBE_HAS_LATEST_TRANSACTION}> ?property_iri . 
+            }}
+        """
+        query = self.remove_unnecessary_whitespace(query)
+        res = self.performQuery(query)
+        if not res:
+            # In case date or price (or both) are missing (i.e. empty SPARQL result), return Nones
+            res = dict(zip(['price', 'date', 'property_iri'], (None,)*3))
+        else:
+            res = res[0]
+            # Cast price to float
+            try:
+                res['price'] = float(res['price'])
+            except:
+                res['price'] = None
+            # Extract relevant date (YYYY-MM) from string
+            try:
+                res['date'] = dt.datetime.strptime(res['date'], TIME_FORMAT_LONG).strftime(TIME_FORMAT_SHORT)
+            except:
+                res['date'] = None
+        return res
+
+
+    def get_floor_area_and_avg_price(self, floor_area_iri:str) -> dict:
+        # Retrieve total floor area and property price index (representative for 
+        # associated property) of given floor area IRI
+        # Returns dictionary with keys: floor_area, avg_price, property_iri
+
+        query = f"""
+            SELECT ?floor_area ?avg_price ?property_iri
+            WHERE {{   
+            <{floor_area_iri}> <{OM_HAS_VALUE}>/<{OM_NUM_VALUE}> ?floor_area ; 
+                               ^<{OBE_HAS_TOTAL_FLOOR_AREA}> ?property_iri . 
+            ?property_iri <{OBE_HAS_ADDRESS}>/<{OBE_HAS_POSTALCODE}> ?postcode_iri . 
+            ?avg_price_iri <{OBE_REPRESENTATIVE_FOR}> ?postcode_iri ; 
+                           <{OM_HAS_VALUE}>/<{OM_NUM_VALUE}> ?avg_price . 
+            }}
+        """
+        query = self.remove_unnecessary_whitespace(query)
+        res = self.performQuery(query)
+        if not res:
+            # In case floor area or average price (or both) are missing (i.e. empty SPARQL result), return Nones
+            res = dict(zip(['floor_area', 'avg_price', 'property_iri'], (None,)*3))
+        else:
+            res = res[0]
+            # Cast floor area and average price to float
+            for key in ['floor_area', 'avg_price']:
+                try:
+                    res[key] = float(res[key])
+                except:
+                    res[key] = None
+        return res
+
+
+    def instantiate_property_value(self, property_iri, property_value_iri, property_value) -> str:
```

Comment:
I would recommend passing in the instance of `rdflib.Graph` to this function and directly adding triples to the graph instance, e.g.
```python
g.add((URIRef(property_value_iri), RDF.type, URIRef(OM_AMOUNT_MONEY)))
```

I found this way to be neater than manually constructing an "insert data" string.

Reply:
Addressed in [5b702ce](https://github.com/cambridge-cares/TheWorldAvatar/pull/454/commits/5b702cea8baab877c9582849fd9e1e162da20a8c)